lib/image_optim/worker.rb

```diff
@@ -151,7 +153,18 @@ def run_command(cmd_args)
           {:out => Path::NULL, :err => Path::NULL},
         ].flatten
       end
-      Cmd.run(*args)
+
+      seconds_to_timeout = timeout || @image_optim.timeout
+
+      if seconds_to_timeout > 0
+        begin
+          Cmd.run_with_timeout(seconds_to_timeout, *args)
+        rescue Cmd::TimeoutExceeded
+          raise ImageOptim::Worker::TimeoutExceeded
```

Comment:
This exception is not handled, cli interface will fail for all images and I would expect timeout to stop only one worker and try next one.